purchases/src/main/kotlin/com/revenuecat/purchases/models/GoogleSubscriptionOption.kt

```diff
@@ -128,5 +128,6 @@ class GoogleSubscriptionOption @JvmOverloads constructor(
             id,
             productDetails,
             offerToken,
+            addOnProducts = emptyList(),
```

Comment:
I believe this is fine because a subscription option shouldn't ever have add-ons, but let me know if there's something here I may be missing!

Reply:
Hmm well... I guess a developer might potentially want to add add-ons even when selecting a specific subscription option to purchase. A subscription option is just a way to select the specific offer that developers want within a base plan but doesn't exclude supporting add-ons IMO...

Reply:
Right! I think I had things twisted around in my head. We should add support for this, so that developers can do something like:

```kotlin
val purchaseParams = PurchaseParams.Builder(activity, subscriptionOption)
      .addOnProducts(...)
      .build()
```

We should also consider having a builder function that is `.addOnSubscriptionOptions()` to go with `.addOnProducts()` and `.addOnPackages()`.

Perhaps this could go in a follow-up PR since this one is already getting a bit large?